mreg_cli/api/abstracts.py

```diff
@@ -95,9 +93,18 @@ def output_timestamps(self, padding: int = 14) -> None:
         output_manager.add_line(f"{'Updated:':<{padding}}{self.updated_at:%c}")
 
 
-class APIMixin(Generic[BMT], ABC):
+class APIMixin(ABC):
     """A mixin for API-related methods."""
 
+    def __init_subclass__(cls) -> None:
+        """Ensure that the subclass inherits from BaseModel."""
+        super().__init_subclass__()
+        if BaseModel not in cls.__mro__ and not cls.__name__.startswith("With"):
```

Comment:
Either way really. It's not *that* expensive to inherit from BaseModel?

Reply:
No, I don't think it's a big deal, since we will always be applying this on `BaseModel` instances anyway. If we run into some very weird MRO issues we can revisit this, but I'll remove the special case for now.